jsonurl.d.ts

```diff
@@ -0,0 +1,64 @@
+interface JsonURLOptions {
+    // These are defined as protected in the comments
+    // setOrDefault(src, key, defValue);
+    // setOverride(src, key);
+    // setOrDefaultInt(src, key, defValue);
+}
+
+export interface JsonURLParseOptions extends JsonURLOptions {
+    //See https://github.com/jsonurl/jsonurl-js/blob/master/src/JsonURLParseOptions.js
+
+    //public properties
+    maxParseDepth?: number; //default 1 << 5 === 32
+    maxParseValues?: number; //default 1 << 12 === 4096
+    maxParseChars?: number; //default 1 << 15 === 32K
+    emptyValue?: any; //default is {}
+    nullValue?: any; //default is null
+
+    //These feel like they are intended to be protected
```

Comment:
I can verify the intention is for them to be public. They allow the caller to modify parse behavior. The comments on the `parse()` method should provide a brief summary and many of them are outlined in section 2.9 of the specification.

Thanks for these comments, BTW! Knowing where the docs fall short will help me know where to improve them.

Reply:
NP.

With this start on the typescript definitions, I think you have a good feel of how to define them.  Feel free to rework them to represent the intended interfaces.  I don't mind if you close this and replace it with another pull request.  I would just like to have some typescript definitions in the near future that represent the public interfaces accurately.

Reply:
@darcyparker let me know if you see any concerns with #177